cpp/util/etcd_test.cc

```diff
@@ -245,78 +256,106 @@ TEST_F(EtcdTest, TestGetAll) {
 }
 
 TEST_F(EtcdTest, TestCreate) {
-  EXPECT_CALL(client_, Generic(kEntryKey,
-                               AllOf(Contains(Pair(kValueParam, "123")),
-                                     Contains(Pair(kPrevExistParam, kFalse))),
-                               UrlFetcher::Verb::PUT, _, _))
-      .WillOnce(Invoke(
-          bind(&GenericReturn, Status::OK, MakeJson(kCreateJson), 1, _4, _5)));
+  EXPECT_CALL(
+      url_fetcher_,
+      Fetch(IsUrlFetchRequest(
+                UrlFetcher::Verb::PUT, URL(GetEtcdUrl(kEntryKey)),
+                ElementsAre(Pair(StrCaseEq("content-type"),
+                                 "application/x-www-form-urlencoded")),
+                "consistent=true&prevExist=false&quorum=true&value=123"),
+            _, _))
+      .WillOnce(
+          Invoke(bind(HandleFetch, Status::OK, 200,
+                      UrlFetcher::Headers{make_pair("x-etcd-index", "1")},
```

Comment:
This is kinda weird in that `X-Etcd-Index` will be less than the `modifiedIndex` of whatever's returned, is that deliberate to ensure we choose the higher of the two?